packages/relay/src/utils.ts

```diff
@@ -55,4 +58,55 @@ export class Utils {
         throw new Error(`Invalid OPERATOR_KEY_FORMAT provided: ${process.env.OPERATOR_KEY_FORMAT}`);
     }
   }
+
+  /**
+   * Generate logs bloom for synthetic transaction
+   * @param address
+   * @param topics
+   */
+  public static createSyntheticLogsBloom(address: string, topics: string[]): string {
+    if (!topics.length) {
+      return EthImpl.emptyBloom;
+    }
+
+    const items = [address, ...topics];
+    const BYTE_SIZE = 256;
+    const MASK = 0x7ff;
+    const bitvector = new Uint8Array(BYTE_SIZE);
+    for (let k = 0; k < items.length; k++) {
+      const item = Buffer.alloc(32, strip0x(keccak256(items[k])), 'hex');
+      for (let i = 0; i < 3; i++) {
+        const first2bytes = new DataView(item.buffer).getUint16(i * 2);
+        const loc = MASK & first2bytes;
+        const byteLoc = loc >> 3;
+        const bitLoc = 1 << loc % 8;
+        bitvector[BYTE_SIZE - byteLoc - 1] |= bitLoc;
+      }
+    }
+
+    return prepend0x(Buffer.from(bitvector).toString('hex'));
+  }
+
+  /**
+   * Check whether an item exists in the hex encoded logs bloom bitvector
+   * @param item
+   * @param bitvector
+   */
+  public static checkInLogsBloom(item: string, bitvector: string): boolean {
```

Comment:
We can either add input validation here:
```typescript
  public static checkInLogsBloom(item: string, bitvector: string): boolean {
    if (!item?.length) {
      throw new Error("Item must be a non-empty string");
    }
    if (!bitvector?.length) {
      throw new Error("Bitvector must be a non-empty string");
    }
    ...
  }
```

or add a safety check and make it return `false` in those cases:
```typescript
  public static checkInLogsBloom(item: string, bitvector: string): boolean {
    if (!item?.length || !bitvector?.length) {
      return false;
    }
    ...
  }
```

Reply:
The method is used only for test purposes so the better place to be is `utils.spec.ts` IMO.